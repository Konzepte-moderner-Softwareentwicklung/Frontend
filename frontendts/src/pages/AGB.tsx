export default function AGB() {

  return (
    <div className="flex items-center justify-center min-h-screen bg-blue-300">
        <div className="max-w-3xl mx-auto p-6 text-gray-800">
        <h1 className="text-3xl font-bold mb-4">Allgemeine Geschäftsbedingungen (AGB)</h1>

        <section className="mb-6">
            <h2 className="text-xl font-semibold mb-2">1. Geltungsbereich</h2>
            <p>
            Diese Allgemeinen Geschäftsbedingungen (AGB) gelten für die Nutzung der App MyCargonaut und alle damit verbundenen Leistungen, die von MyCargonaut bereitgestellt werden.
            </p>
        </section>

        <section className="mb-6">
            <h2 className="text-xl font-semibold mb-2">2. Vertragspartner</h2>
            <p>
            Der Vertrag kommt zustande mit:
            <br />
            MyCargonaut<br />
            Max Mustermann<br />
            Musterstraße 123<br />
            12345 Musterstadt<br />
            Deutschland
            </p>
        </section>

        <section className="mb-6">
            <h2 className="text-xl font-semibold mb-2">3. Vertragsgegenstand</h2>
            <p>
            MyCargonaut stellt eine Plattform zur Verfügung, über die Nutzer Fahrgemeinschaften organisieren und Fahrten anbieten oder buchen können.
            </p>
        </section>

        <section className="mb-6">
            <h2 className="text-xl font-semibold mb-2">4. Registrierung und Nutzerkonto</h2>
            <p>
            Die Nutzung von MyCargonaut setzt eine Registrierung und Erstellung eines Nutzerkontos voraus. Die bei der Registrierung angegebenen Daten müssen vollständig und korrekt sein.
            </p>
        </section>

        <section className="mb-6">
            <h2 className="text-xl font-semibold mb-2">5. Pflichten der Nutzer</h2>
            <p>
            Nutzer verpflichten sich, keine rechtswidrigen Inhalte über die Plattform zu verbreiten und die Rechte Dritter zu achten. 
            Zudem dürfen Fahrten nur angeboten werden, wenn sie tatsächlich durchgeführt werden können.
            </p>
        </section>

        <section className="mb-6">
            <h2 className="text-xl font-semibold mb-2">6. Haftung</h2>
            <p>
            MyCargonaut haftet nicht für das Verhalten der Nutzer oder für Schäden, die aus der Nutzung der Plattform entstehen, 
            es sei denn, diese beruhen auf grober Fahrlässigkeit oder Vorsatz seitens MyCargonaut.
            </p>
        </section>

        <section className="mb-6">
            <h2 className="text-xl font-semibold mb-2">7. Vertragslaufzeit und Kündigung</h2>
            <p>
            Der Nutzungsvertrag wird auf unbestimmte Zeit geschlossen und kann von beiden Seiten jederzeit ohne Angabe von Gründen gekündigt werden.
            </p>
        </section>

        <section className="mb-6">
            <h2 className="text-xl font-semibold mb-2">8. Änderungen der AGB</h2>
            <p>
            MyCargonaut behält sich das Recht vor, diese AGB jederzeit zu ändern. Die Nutzer werden über Änderungen rechtzeitig informiert.
            </p>
        </section>

        <section className="mb-6">
            <h2 className="text-xl font-semibold mb-2">9. Anwendbares Recht</h2>
            <p>
            Es gilt das Recht der Bundesrepublik Deutschland. Gerichtsstand ist, soweit gesetzlich zulässig, der Sitz von MyCargonaut.
            </p>
        </section>

        <section>
            <p>
            Stand: Juni 2025
            </p>
        </section>
        </div>
    </div>
  );
};