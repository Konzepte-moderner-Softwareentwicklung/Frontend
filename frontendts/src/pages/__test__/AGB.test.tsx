// src/pages/__test__/AGB.test.tsx
import { render, screen, within } from '@testing-library/react';
import AGB from '../AGB'; // <— anpassen falls Pfad anders

describe('AGB page', () => {
  test('rendert Hauptüberschrift und Einleitung', () => {
    render(<AGB />);

    const h1 = screen.getByRole('heading', {
      level: 1,
      name: /Allgemeine Geschäftsbedingungen \(AGB\)/i,
    });
    expect(h1).toBeInTheDocument();

    expect(screen.getByText(/Stand:\s*Juni 2025/i)).toBeInTheDocument();
  });

  test('enthält alle Abschnittsüberschriften 1–9', () => {
    render(<AGB />);

    const h2s = screen.getAllByRole('heading', { level: 2 });
    expect(h2s).toHaveLength(9);

    expect(
      screen.getByRole('heading', { level: 2, name: /^1\.\s*Geltungsbereich$/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { level: 2, name: /^2\.\s*Vertragspartner$/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { level: 2, name: /^3\.\s*Vertragsgegenstand$/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { level: 2, name: /^4\.\s*Registrierung und Nutzerkonto$/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { level: 2, name: /^5\.\s*Pflichten der Nutzer$/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { level: 2, name: /^6\.\s*Haftung$/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { level: 2, name: /^7\.\s*Vertragslaufzeit und Kündigung$/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { level: 2, name: /^8\.\s*Änderungen der AGB$/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { level: 2, name: /^9\.\s*Anwendbares Recht$/i })
    ).toBeInTheDocument();
  });

  test('Layout-Container vorhanden (Smoke-Test, keine harten CSS-Assertions)', () => {
    render(<AGB />);
    const outer = screen.getByRole('heading', {
      level: 1,
      name: /Allgemeine Geschäftsbedingungen/i,
    }).closest('div');
    expect(outer).toBeTruthy();
  });

  test('semantische Struktur: jede H2 hat mindestens einen Paragraphen im Section-Container', () => {
    render(<AGB />);
    const h2s = screen.getAllByRole('heading', { level: 2 });

    for (const h2 of h2s) {
      const section = h2.closest('section') as HTMLElement | null;
      expect(section).toBeTruthy();

      if (section) {
        const paragraphs = Array.from(section.querySelectorAll('p'));
        expect(paragraphs.length).toBeGreaterThanOrEqual(1);
        // Optional: Prüfe, dass der erste Paragraph nicht leer ist
        expect(paragraphs[0].textContent?.trim().length).toBeGreaterThan(0);
      }
    }
  });

  test('Snapshot der Seite (stabil bei statischem Inhalt)', () => {
    const { container } = render(<AGB />);
    expect(container).toMatchSnapshot();
  });
});
