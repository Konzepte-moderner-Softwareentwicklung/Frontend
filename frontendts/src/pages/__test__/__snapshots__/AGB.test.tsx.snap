// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AGB page Snapshot der Seite (stabil bei statischem Inhalt) 1`] = `
<div>
  <div
    class="flex items-center justify-center min-h-screen bg-blue-300"
  >
    <div
      class="max-w-3xl mx-auto p-6 text-gray-800"
    >
      <h1
        class="text-3xl font-bold mb-4"
      >
        Allgemeine Geschäftsbedingungen (AGB)
      </h1>
      <section
        class="mb-6"
      >
        <h2
          class="text-xl font-semibold mb-2"
        >
          1. Geltungsbereich
        </h2>
        <p>
          Diese Allgemeinen Geschäftsbedingungen (AGB) gelten für die Nutzung der App MyCargonaut und alle damit verbundenen Leistungen, die von MyCargonaut bereitgestellt werden.
        </p>
      </section>
      <section
        class="mb-6"
      >
        <h2
          class="text-xl font-semibold mb-2"
        >
          2. Vertragspartner
        </h2>
        <p>
          Der Vertrag kommt zustande mit:
          <br />
          MyCargonaut
          <br />
          Max Mustermann
          <br />
          Musterstraße 123
          <br />
          12345 Musterstadt
          <br />
          Deutschland
        </p>
      </section>
      <section
        class="mb-6"
      >
        <h2
          class="text-xl font-semibold mb-2"
        >
          3. Vertragsgegenstand
        </h2>
        <p>
          MyCargonaut stellt eine Plattform zur Verfügung, über die Nutzer Fahrgemeinschaften organisieren und Fahrten anbieten oder buchen können.
        </p>
      </section>
      <section
        class="mb-6"
      >
        <h2
          class="text-xl font-semibold mb-2"
        >
          4. Registrierung und Nutzerkonto
        </h2>
        <p>
          Die Nutzung von MyCargonaut setzt eine Registrierung und Erstellung eines Nutzerkontos voraus. Die bei der Registrierung angegebenen Daten müssen vollständig und korrekt sein.
        </p>
      </section>
      <section
        class="mb-6"
      >
        <h2
          class="text-xl font-semibold mb-2"
        >
          5. Pflichten der Nutzer
        </h2>
        <p>
          Nutzer verpflichten sich, keine rechtswidrigen Inhalte über die Plattform zu verbreiten und die Rechte Dritter zu achten. Zudem dürfen Fahrten nur angeboten werden, wenn sie tatsächlich durchgeführt werden können.
        </p>
      </section>
      <section
        class="mb-6"
      >
        <h2
          class="text-xl font-semibold mb-2"
        >
          6. Haftung
        </h2>
        <p>
          MyCargonaut haftet nicht für das Verhalten der Nutzer oder für Schäden, die aus der Nutzung der Plattform entstehen, es sei denn, diese beruhen auf grober Fahrlässigkeit oder Vorsatz seitens MyCargonaut.
        </p>
      </section>
      <section
        class="mb-6"
      >
        <h2
          class="text-xl font-semibold mb-2"
        >
          7. Vertragslaufzeit und Kündigung
        </h2>
        <p>
          Der Nutzungsvertrag wird auf unbestimmte Zeit geschlossen und kann von beiden Seiten jederzeit ohne Angabe von Gründen gekündigt werden.
        </p>
      </section>
      <section
        class="mb-6"
      >
        <h2
          class="text-xl font-semibold mb-2"
        >
          8. Änderungen der AGB
        </h2>
        <p>
          MyCargonaut behält sich das Recht vor, diese AGB jederzeit zu ändern. Die Nutzer werden über Änderungen rechtzeitig informiert.
        </p>
      </section>
      <section
        class="mb-6"
      >
        <h2
          class="text-xl font-semibold mb-2"
        >
          9. Anwendbares Recht
        </h2>
        <p>
          Es gilt das Recht der Bundesrepublik Deutschland. Gerichtsstand ist, soweit gesetzlich zulässig, der Sitz von MyCargonaut.
        </p>
      </section>
      <section>
        <p>
          Stand: Juni 2025
        </p>
      </section>
    </div>
  </div>
</div>
`;
